### PR TITLE
ArduCopter: moved hori/vert land control fns to Mode class

### PR DESCRIPTION
**Pull request by EbinPhilip**
- created inner class Land in Mode
- moved land_run_horizontal_control() and land_run_vertical_control() to inner-class Land in Mode
@peterbarker @rmackay9 Made the changed as suggested in PR #7954. Could you please take a look?

### CODE REVIEW

**Comment by EbinPhilip** [comment]
@peterbarker did that. Is it fine now?

File: ArduCopter/mode_land.cpp [hunk 1]
```diff
@@ -136,13 +136,38 @@ void Copter::ModeLand::nogps_run()
...
+	g2(copter.g2),
+	wp_nav(copter.wp_nav),
+	pos_control(copter.pos_control),
+	inertial_nav(copter.inertial_nav),
+	attitude_control(copter.attitude_control),
+	motors(copter.motors),
+	channel_yaw(copter.channel_yaw),
+	G_Dt(copter.G_Dt),
+	ap(copter.ap),
+    ekfGndSpdLimit(copter.ekfGndSpdLimit),
```
Comment (peterbarker) [inline ArduCopter/mode_land.cpp]: whitespace issue

File: ArduCopter/mode_land.cpp [hunk 2]
```diff
@@ -168,7 +193,7 @@ void Copter::land_run_vertical_control(bool pause_descent)
     // compute desired velocity
     const float precland_acceptable_error = 15.0f;
     const float precland_min_descent_speed = 10.0f;
-    int32_t alt_above_ground = mode_land.get_alt_above_ground();
+    int32_t alt_above_ground = get_alt_above_ground();
```
Comment (peterbarker) [inline ArduCopter/mode_land.cpp]: `const` this variable

File: ArduCopter/mode_land.cpp [hunk 3]
```diff
@@ -188,7 +213,7 @@ void Copter::land_run_vertical_control(bool pause_descent)
         // Constrain the demanded vertical velocity so that it is between the configured maximum descent speed and the configured minimum descent speed.
         cmb_rate = constrain_float(cmb_rate, max_land_descent_velocity, -abs(g.land_speed));
 
-        if (doing_precision_landing && rangefinder_alt_ok() && rangefinder_state.alt_cm > 35.0f && rangefinder_state.alt_cm < 200.0f) {
+        if (doing_precision_landing && copter.rangefinder_alt_ok() && copter.rangefinder_state.alt_cm > 35.0f && copter.rangefinder_state.alt_cm < 200.0f) {
```
Comment (peterbarker) [inline ArduCopter/mode_land.cpp]: Not your fault, really, but this line just got nastier :-)

File: ArduCopter/mode_land.cpp [hunk 4]
```diff
@@ -235,19 +260,19 @@ void Copter::land_run_horizontal_control()
...
+        target_yaw_rate = copter.get_pilot_desired_yaw_rate(channel_yaw->get_control_in());
     }
 
 #if PRECISION_LANDING == ENABLED
     bool doing_precision_landing = !ap.land_repo_active && precland.target_acquired();
     // run precision landing
     if (doing_precision_landing) {
         Vector2f target_pos, target_vel_rel;
-        if (!precland.get_target_position_cm(target_pos)) {
+        if (!copter.precland.get_target_position_cm(target_pos)) {
```
Comment (peterbarker) [inline ArduCopter/mode_land.cpp]: We took references for a reason :-)

**Comment by EbinPhilip** [comment]
Sorry, I got a bit careless there. I'll fix it asap

**Comment by EbinPhilip** [comment]
@peterbarker Made those changes

**Comment by EbinPhilip** [comment]
@peterbarker moved those functions back to mode. I couldn't const the precland reference, because precland.target_acquired() changes a bool variable in that class: https://github.com/ArduPilot/ardupilot/blob/6652dd73e4c97f83325a85f3a8e321abf6eed2f2/libraries/AC_PrecLand/AC_PrecLand.cpp#L177

**Comment by peterbarker** [comment]
This is looking good.

Please have two commits, 'though.  One which changes namespace and the second which moves the code from one file to another.  You can mark the second commit as "NFC" - just make sure you don't actually do anything except move code in it!

**Comment by EbinPhilip** [comment]
@peterbarker I've split it into 2 commits. Hope its ok now

**Comment by rmackay9** [comment]
Darn, I'm afraid this needs to be rebased on master.  I tried to overcome the merge conflicts myself but it's not completely trivial so I better leave it to @EbinPhilip.

**Comment by EbinPhilip** [comment]
@rmackay9 I'll do it

**Comment by OXINARF** [comment]
I've rebased, changed the second commit message to end with _(NFC)_ instead of starting with it and merged. Thanks!

**Pull request closed without merging**